test/dlc_tests/conftest.py

```diff
@@ -92,20 +92,31 @@ def ec2_instance(
 ):
     print(f"Creating instance: CI-CD {ec2_key_name}")
     key_filename = test_utils.generate_ssh_keypair(ec2_client, ec2_key_name)
-    instances = ec2_resource.create_instances(
-        KeyName=ec2_key_name,
-        ImageId=ec2_instance_ami,
-        InstanceType=ec2_instance_type,
-        IamInstanceProfile={"Name": ec2_instance_role_name},
-        TagSpecifications=[
-            {
-                "ResourceType": "instance",
-                "Tags": [{"Key": "Name", "Value": f"CI-CD {ec2_key_name}"}]
-            },
-        ],
-        MaxCount=1,
-        MinCount=1,
-    )
+    params = {
+        "KeyName": ec2_key_name,
+        "ImageId": ec2_instance_ami,
+        "InstanceType": ec2_instance_type,
+        "IamInstanceProfile": {"Name": ec2_instance_role_name},
+        "TagSpecifications": [
+                {
+                    "ResourceType": "instance",
+                    "Tags": [{"Key": "Name", "Value": f"CI-CD {ec2_key_name}"}]
+                },
+            ],
+        "MaxCount": 1,
+        "MinCount": 1
+    }
+    extra_volume_size_mapping = [
+                        {
+                            'DeviceName': '/dev/sda1',
+                            'Ebs': {
+                                'VolumeSize': 300,
+                            }
+                        }
+                    ]
+    if "performance" in ec2_key_name and "mxnet" in ec2_key_name and "training_cpu" not in ec2_key_name:
```

Comment:
This relies heavily on the 'ec2_key_name' not changing it's format. Can the same information be obtained from another param? e.g. `request`?

Reply:
Addressed in latest commit